design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,219 @@
+# Proposal to separate resource filter into cluster scope and namespace scope.
+
+## Abstract
+The current filter (IncludedResources/ExcludedResources + IncludeClusterResources flag) is not enough for some special cases, e.g. all namespaced resources + some kind of cluster resource and all namespaced resources + cluster resource excludes.
+Propose to separate `IncludedResources` into two cluster scope and namespace scope parameters, and same change for `ExcludedResources`.
+
+## Background
+There are two sets of resource filters for Velero: `IncludedNamespaces/ExcludedNamespaces` and `IncludedResources/ExcludedResources`. 
+`IncludedNamespaces` means only including the resource types specified in the parameter.
+The k8s resources are separated into cluster scope and namespaced scope.
+As a result, it's hard to include all resources in one group and only including specified resource in other group.
+
+## Goals
+- Make the existing resource filters supporting more scenarios without conflict.
+
+## Non Goals
+- Enrich the resource filtering rules.
+
+
+## High-Level Design
+Two new parameters are added into command `velero backup create`: `--include-cluster-scope-resources` and `--exclude-cluster-scope-resources`. 
+They are used to filter cluster scope resources included or excluded in backup.
+`--include-cluster-scope-resources` valid value includes `*` and comma separated string.
+`--exclude-cluster-scope-resources` valid value is comma separated string.
+
+### Using scenarios:
+#### some namespaced resources + no additional cluster resources
+``` bash
+velero backup create <backup-name>
+    --include-namespaces=default,kube-system
+```
+
+#### some namespaced resources + some additional cluster resources
+``` bash
+velero backup create <backup-name>
+    --include-namespaces=default,kube-system
+    --include-cluster-scope-resources=StorageClass
+```
+
+#### some namespaced resources + all cluster resources
+``` bash
+velero backup create <backup-name>
+    --include-namespaces=default,kube-system
+    --include-cluster-scope-resources=*
+```
+
+or
+
+``` bash
+velero backup create <backup-name>
+    --include-namespaces=default,kube-system
+    --include-cluster-resources=true
+```
+
+#### all namespaced resources + no additional cluster resources
+``` bash
+velero backup create <backup-name>
+    --include-namespaces=*
+```
```

Comment:
include-cluster-resources would be nil here, so you'd pull in only relevant cluster resources (PVs, etc.)